spec/intro.dd

```diff
@@ -10,59 +10,58 @@ to create native libraries or executables.)
 $(P This document is the reference manual for the D Programming Language. For more information and
 other documents$(COMMA) see $(LINK2 https://dlang.org/, The D Language Website).)
 
-$(H2 Phases of Compilation)
+$(H2 Conceptual models)
 
-$(P The process of compiling is divided into multiple phases. Each phase has no
-dependence on subsequent phases. For example, the scanner is not perturbed by
-the semantic analyzer. This separation of the passes makes language tools like
-syntax directed editors relatively easy to produce. It also is possible to
-compress D source by storing it in $(SINGLEQUOTE tokenized) form.)
+$(H3 Environments)
 
-$(OL
-        $(LI $(B source character set)$(BR)
+$(P The data-processing-system in which a D program is translated from source to executable format
+will be called the translation environment and the data-processing-system in which the translated
+D program is executed will be called the execution environment. The translation environment and
+the execution environment need not have the same system architecture.)
 
-        The source file is checked to see what character set it is,
-        and the appropriate scanner is loaded. ASCII and UTF
-        formats are accepted.
-        )
+$(H3 Program Structure)
 
-        $(LI $(B script line) $(BR)
+$(P A D program is kept in units called source files. Each source file contains a unit of translation
+called the module. D modules need not all be translated at the same time. Previously translated
+D modules may be individually preserved or in libraries. The separate modules of a program communicate
+by calls to functions whose identifers have external linkage, manipulation of objects whose identifers
+have external linkage, or manipulation of data files. Modules may be separately translated and then
+later linked to produce an executable program.)
```

Comment:
- s/the module/a module
- No need to say "D modules" every time. It's obvious.
- What does "individually preserved" mean?
- s/The separate modules of a program communicate by calls/Modules communicate via
- What is manipulation of data files?

Reply:
Revised, please have a look.